lib/rspec/support/differ.rb

```diff
@@ -116,6 +115,12 @@ def build_hunks(actual, expected)
         HunkGenerator.new(actual, expected).hunks
       end
 
+      def final_line(last_hunk)
+        output_message = last_hunk.diff(format_type).to_s.strip
+        # Checks last char of string for + or -, appends message on match
+        output_message.gsub(/(?<=[+-]$)\z/, "\\ No newline at end of input")
```

Comment:
I'm up for giving it another go! By 'old logic' do you mean 
```
def expected_lines
  @expected.to_s.split("\n", -1).map! { |e| e.chomp }
end
```
in https://github.com/ChaeOkay/rspec-support/blob/b4a3bacd999429e7d64b88486e86597eefdbbb14/lib/rspec/support/hunk_generator.rb#L27 ?